lib/tools.js

```diff
@@ -1997,6 +2005,33 @@ function removeIdFromAllEnums(objects, id) {
     });
 }
 
+/**
+ * Parses dependencies to standardized object of form
+ *
+ * @alias parseDependencies
+ * @memberof tools
+ * @param {string[]|object[]|string} dependencies dependencies array or single dependency
+ * @returns {object} parsed dependencies
+ */
+function parseDependencies(dependencies) {
+    const adapters = {};
+    if (Array.isArray(dependencies)) {
+        dependencies.forEach(rule => {
+            if (typeof rule === 'string') {
+                // No version given, all are okay
+                adapters[rule] = '*';
+            } else if (isObject(rule)) {
+                // can be object containing single adapter or multiple
+                Object.keys(rule).filter(adapter => !adapters[adapter]).forEach(adapter => adapters[adapter] = rule[adapter]);
+            }
+        });
+    } else if (typeof dependencies === 'string') {
+        // its a single string without version requirement
+        adapters[dependencies] = '*';
+    }
```

Comment:
If we add another case that checks for object and return it then we are compatible to the setup* cases before